File: android/ac-charts/src/main/kotlin/com/microsoft/adaptivecards/charts/ChartModels.kt
package com.microsoft.adaptivecards.charts

import androidx.compose.ui.graphics.Color

object ChartColors {
    val defaultPalette = listOf(
        Color(0xFF0078D4), // Blue
        Color(0xFF00BCF2), // Cyan
        Color(0xFF8764B8), // Purple
        Color(0xFF00B7C3), // Teal
        Color(0xFFFFB900), // Yellow
        Color(0xFFD83B01), // Orange
        Color(0xFFE74856), // Red
        Color(0xFF00CC6A)  // Green
    )
    
    fun colors(from: List<String>?): List<Color> {
        return from?.mapNotNull { parseHexColor(it) } ?: defaultPalette
    }
    
    private fun parseHexColor(hex: String): Color? {
        val cleanHex = hex.removePrefix("#")
        return try {
            when (cleanHex.length) {
                6 -> Color(0xFF000000 or cleanHex.toLong(16))
                8 -> Color(cleanHex.toLong(16))
                else -> null
            }
        } catch (e: Exception) {
            null
        }
    }
}

enum class ChartSize(val heightDp: Int) {
    SMALL(150),
    MEDIUM(250),
    LARGE(350),
    AUTO(250);
    
    companion object {
        fun from(string: String?): ChartSize {
            return when (string?.lowercase()) {
                "small" -> SMALL
                "medium" -> MEDIUM
                "large" -> LARGE
                else -> AUTO
            }
        }
    }
}

File: ios/Sources/ACCharts/ChartModels.swift
import SwiftUI
import ACFluentUI

public struct ChartColors {
    public static let defaultPalette: [Color] = [
        Color(hex: "#0078D4"), // Blue
        Color(hex: "#00BCF2"), // Cyan
        Color(hex: "#8764B8"), // Purple
        Color(hex: "#00B7C3"), // Teal
        Color(hex: "#FFB900"), // Yellow
        Color(hex: "#D83B01"), // Orange
        Color(hex: "#E74856"), // Red
        Color(hex: "#00CC6A")  // Green
    ]

    public static func colors(from hexColors: [String]?) -> [Color] {
        if let hexColors = hexColors, !hexColors.isEmpty {
            return hexColors.map { Color(hex: $0) }
        }
        return defaultPalette
    }
}

public enum ChartSize {
    case small
    case medium
    case large
    case auto

    public var height: CGFloat {
        switch self {
        case .small: return 150
        case .medium: return 250
        case .large: return 350
        case .auto: return 250
        }
    }

    public static func from(_ string: String?) -> ChartSize {
        guard let string = string else { return .auto }
        switch string.lowercased() {
        case "small": return .small
        case "medium": return .medium
        case "large": return .large
        default: return .auto
        }
    }
}
